### PR TITLE
efibootmgr: new package

### PR DESCRIPTION
**Pull request by oskarirauta**
Maintainer: me / @oskarirauta 
Compile tested: x86-64, recent git
Run tested: x86-64, recent git

Description:
Linux user-space application to modify the Intel Extensible Firmware Interface (EFI) Boot 
Manager. This application can create and destroy boot entries, change the boot order, change the next running 
boot option, and more.

Depends on my another PR #20276 (efivar library)

Signed-off-by: Oskari Rauta <oskari.rauta@gmail.com>

### CODE REVIEW

**Comment by oskarirauta** [comment]
Build ofcourse won't work until #20276 is merged, but could someone please help a bit here, what are these ci errors that I am getting with nearly every PR that I've recently made?

**Comment by oskarirauta** [comment]
@1715173329 

This looks good to go too, maybe check this too after efivar library..? :)
This is the utility used by most distributions to modify boot order/options with efi.

File: utils/efibootmgr/Makefile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+PKG_SOURCE_DATE:=2022-11-12
+PKG_SOURCE_VERSION:=1904f9cd5a48207d49e393ac8de824f54ccfb697
+PKG_MIRROR_HASH:=0355e00bb54d468ecbaa106aa703dd389f2f2f4c7b7afb78a258cec10d75f78d
+
+PKG_LICENSE:=GPL-2.0-only
+PKG_LICENSE_FILES:=COPYING
+PKG_BUILD_PARALLEL:=1
+PKG_INSTALL:=1
+
+PKG_MAINTAINER:=OSkari Rauta <oskari.rauta@gmail.com>
```
Comment (1715173329) [inline utils/efibootmgr/Makefile:18]:
```suggestion
PKG_LICENSE:=GPL-2.0-only
PKG_LICENSE_FILES:=COPYING
PKG_MAINTAINER:=OSkari Rauta <oskari.rauta@gmail.com>

PKG_BUILD_PARALLEL:=1
PKG_INSTALL:=1
```

File: utils/efibootmgr/Makefile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  Manager. This application can create and destroy boot entries, change the boot order, change the next running 
+  boot option, and more.
+
+  Details on the EFI Boot Manager are available from the EFI Specification, v1.02 or above, available from:
+  http://www.uefi.org
+
+  Note: efibootmgr requires either the efivarfs or the legacy efivars kernel module to be loaded prior to use.
+endef
+
+MAKE_VARS += EFIDIR="/boot/EFI" PCDIR="$(STAGING_DIR)/usr/lib/pkgconfig" EXTRAINCDIR="-I$(STAGING_DIR)/usr/include -I$(STAGING_DIR)/usr/include/efivar" EXTRALIBDIR="-L$(STAGING_DIR)/usr/lib"
```
Comment (1715173329) [inline utils/efibootmgr/Makefile:41]:
```suggestion
MAKE_VARS += \
	EFIDIR="/boot/EFI" \
	PCDIR="$(STAGING_DIR)/usr/lib/pkgconfig" \
	EXTRAINCDIR="-I$(STAGING_DIR)/usr/include -I$(STAGING_DIR)/usr/include/efivar" \
	EXTRALIBDIR="-L$(STAGING_DIR)/usr/lib"
```

File: utils/efibootmgr/Makefile [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+PKG_INSTALL:=1
+
+PKG_MAINTAINER:=OSkari Rauta <oskari.rauta@gmail.com>
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/efibootmgr
+  SECTION:=utils
+  CATEGORY:=Utilities
+  TITLE:=Linux user-space application to modify the Intel Extensible Firmware Interface (EFI) Boot Manager
```
Comment (1715173329) [inline utils/efibootmgr/Makefile:25]: I think it would be too long for menuconfig, can you try to shorten it?

File: utils/efibootmgr/Makefile [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  DEPENDS:=@TARGET_x86_64 +efivar +libpopt
+  URL:=https://github.com/rhboot/efibootmgr
+endef
+
+define Package/efibootmgr/description
+  This is efibootmgr, a Linux user-space application to modify the Intel Extensible Firmware Interface (EFI) Boot
```
Comment (1715173329) [inline utils/efibootmgr/Makefile:31]: please wrap a new line after every 70 characters.

**Review by 1715173329 — CHANGES_REQUESTED** [review]

File: utils/efibootmgr/Makefile (hunk 3 above)
Comment (oskarirauta) [reply utils/efibootmgr/Makefile:25]: Sure thing

File: utils/efibootmgr/Makefile (hunk 4 above)
Comment (oskarirauta) [reply utils/efibootmgr/Makefile:31]: Consider it as done :)

**Comment by oskarirauta** [comment]
@1715173329 

This also says that change request is pending- but as far as I know, all requsted changes should now be resolved. Could you verify if I have missed one?

**Comment by oskarirauta** [comment]
Once efivar is merged, this should be possible.

**Review by 1715173329 — APPROVED** [review]

**Comment by oskarirauta** [comment]
refind_screenshot

Screenshot of rEFInd, installed as a (default) boot option with efibootmgr.

**Pull request merged**